dracut/repartition/endless-repartition.sh

```diff
@@ -59,13 +59,13 @@ get_last_char() {
 }
 
 # We have to work with a variety of setups here:
-# - Either SCSI or MMC block devices with different naming conventions
+# - SCSI/MMC/NVMe block devices with different naming conventions
 # - A different partition numbering scheme as not all configurations have
 #   a ESP and a BIOS boot partition.
 partno=$(get_last_char ${root_part})
 swap_partno=$((partno + 1))
 case ${root_part} in
-  /dev/mmcblk?p?)
+  /dev/*p[0-9])
```

Comment:
I suppose *technically* if you had 16 ATA devices, `/dev/sdp3` would match this branch and things would go wrong; so the `sd??` match should come before this more-lenient match. I don't think this is especially likely…